### PR TITLE
Use the supplied contact name in JavaMail

### PR DESCRIPTION
**Pull request by timyates**
As reported in Gitter, we were ignoring the supplied Contact name if one was provided, and just sending emails to (and from) the email address.

This check for a supplied name, and uses it to create the InternetAddress if one were specified

### CODE REVIEW

File: email-javamail/src/test/groovy/io/micronaut/email/javamail/JavaMailBodyAndAttachmentSpec.groovy [hunk 1]
```diff
@@ -51,8 +52,8 @@ class JavaMailBodyAndAttachmentSpec extends Specification {
         MailhogClient client = applicationContext.getBean(MailhogClient)
 
         and:
-        String from = "sender@here.com"
-        String to = "receiver@here.com"
+        Contact from = new Contact("sender@here.com", "Zapp Brannigan")
+        Contact to = new Contact("receiver@here.com", "Kif")
```
Comment (sdelamo) [inline email-javamail/src/test/groovy/io/micronaut/email/javamail/JavaMailBodyAndAttachmentSpec.groovy:56]: do we have a test which verifies it works without passing a name?

Comment (timyates) [reply email-javamail/src/test/groovy/io/micronaut/email/javamail/JavaMailBodyAndAttachmentSpec.groovy:56]: No, I'll add one now

Comment (timyates) [reply email-javamail/src/test/groovy/io/micronaut/email/javamail/JavaMailBodyAndAttachmentSpec.groovy:96]: Pushed a tests here https://github.com/micronaut-projects/micronaut-email/pull/59/commits/691f1026de7247a5447488bf0ede38c32a1f7827 🤞

Comment (sdelamo) [reply email-javamail/src/test/groovy/io/micronaut/email/javamail/JavaMailBodyAndAttachmentSpec.groovy:96]: thanks

**Pull request merged**